src/sentry/notifications/notificationcontroller.py

```diff
@@ -54,19 +59,39 @@ class NotificationController:
 
     def __init__(
         self,
-        recipients: Iterable[RpcActor] | Iterable[Team] | Iterable[RpcUser] | Iterable[User],
+        recipients: Iterable[Recipient],
         project_ids: Iterable[int] | None = None,
         organization_id: int | None = None,
         type: NotificationSettingEnum | None = None,
         provider: ExternalProviderEnum | None = None,
     ) -> None:
-        self.recipients = recipients
+        org = (
+            serialize_organization_mapping(
+                OrganizationMapping.objects.filter(organization_id=organization_id).first()
+            )
+            if organization_id
+            else None
+        )
+        if org and features.has("organizations:team-workflow-notifications", org):
+            self.recipients: list[Recipient] = []
+            for recipient in recipients:
+                if recipient_is_team(
+                    recipient
+                ):  # this call assures the recipient type is okay (so can safely ignore below type errors)
+                    if team_is_valid_recipient(recipient):  # type: ignore
+                        self.recipients.append(recipient)
+                    else:
+                        self.recipients += get_team_members(recipient)  # type: ignore
+                else:
+                    self.recipients.append(recipient)
+        else:
+            self.recipients = list(recipients)
```

Comment:
@isabellaenriquez nit: can we move this below `self.provider = provider`?

Reply:
you mean move the whole logic for setting self.recipients down or just the case where the feature flag is off?